libimage/pull.go

```diff
@@ -343,6 +324,65 @@ func (r *Runtime) copyFromDockerArchiveReaderReference(ctx context.Context, read
 	return destNames, nil
 }
 
+func (r *Runtime) copyFromOCIArchive(ctx context.Context, readerRef types.ImageReference, options *CopyOptions) ([]string, error) {
+	reader, err := ociArchiveTransport.NewReader(ctx, &r.systemContext, readerRef)
+	if err != nil {
+		return nil, err
+	}
+	return r.copyFromOCIArchiveReader(ctx, reader, options)
```

Comment:
If `readerRef` points at specific image, this seems to completely lose that choice.

Cc: @vrothberg . I thought the distinction is that `podman pull oci-archive:…` always pulls exactly _the one_ image that c/image uses under that reference, while `podman load` can possibly pull _all_ images from an archive.

(Compare also the discussion about `NewReaderForReference` / `LoadManifestDescriptor` in the c/image PR.)

Reply:
> Cc: @vrothberg . I thought the distinction is that podman pull oci-archive:… always pulls exactly the one image that c/image uses under that reference, while podman load can possibly pull all images from an archive.

Yes, `podman pull` expects one image.  You can check out the behavior via `podman pull docker-archive:libimage/testdata/docker-two-images.tar.xz`